src/profile-logic/tracks.js

```diff
@@ -543,26 +543,69 @@ export function getLocalTrackName(
   }
 }
 
-// Any thread with less than 1% non-idle time will be hidden.
-const PERCENTAGE_ACTIVE_SAMPLES = 0.01;
-
 /**
  * Determine if a thread is idle, so that it can be hidden. It is really annoying for an
- * end user to load a profile full of empty and idle threads. This function goes through
- * all of the samples in the thread, and sees if some large percentage of them are idle.
+ * end user to load a profile full of empty and idle threads. This function uses
+ * various rules to determine if a thread is idle.
  */
 function _isThreadIdle(profile: Profile, thread: Thread): boolean {
   if (
     // Don't hide the compositor.
     thread.name === 'Compositor' ||
-    // Don't hide the main thread.
+    // Don't hide the main thread of the parent process.
     (thread.name === 'GeckoMain' && thread.processType === 'default') ||
     // Don't hide the GPU thread on Windows.
     (thread.name === 'GeckoMain' && thread.processType === 'gpu')
   ) {
     return false;
   }
 
+  if (_isContentThreadWithNoPaint(thread)) {
+    return true;
+  }
+
+  return _isThreadMostlyFullOfIdleSamples(profile, thread);
```

Comment:
I'm putting that possibility here:
```javascript
return _isContentThreadWithNoPaint(thread) || _isThreadMostlyFullOfIdleSamples(profile, thread);
```

but I'm not sure this is more readable that what you wrote so let's keep what you did.